directives/questionnaire.py

```diff
@@ -120,6 +122,22 @@ def run(self):
                 raise SphinxError(u'Number of fields to sample randomly should greater than zero.')
             data[u'fieldgroups'][0]['pick_randomly'] = pick_randomly
 
+        if 'allow-assistant-grading' in self.options:
```

Comment:
Since the directive classes (Questionnaire and SubmitForm) inherit from AbstractExercise, you could move the identical code from the two classes into a new method in the super class. Then you just call the method when you parse the options `allow-assistant-viewing` and `allow-assistant-grading`.